supervision/detection/line_zone.py

```diff
@@ -158,28 +158,29 @@ def trigger(self, detections: Detections) -> Tuple[np.ndarray, np.ndarray]:
             ]
         )
 
+        cross_products_1 = self._cross_product(all_anchors, self.limits[0])
+        cross_products_2 = self._cross_product(all_anchors, self.limits[1])
+        # anchor is in limits if it's on the same side of both limit vectors
+        in_limits = ~np.logical_xor(cross_products_1 > 0, cross_products_2 > 0)
+        # Reduce array to find out if all anchors for a detection are within limits
+        in_limits = np.min(in_limits, axis=0)
+
+        # Calculate which anchors lie to the left of the line
+        triggers = self._cross_product(all_anchors, self.vector) < 0
+        # Reduce to find out if all anchors for a
+        # detection lie to the left (or right) of the line
+        max_triggers = np.max(triggers, axis=0)
```

Comment:
Hi @tc360950 👋🏻 

Could you explain what is the interpretation of:

```python
max_triggers = np.max(triggers, axis=0)
min_triggers = np.min(triggers, axis=0)

...

if min_triggers[i] != max_triggers[i]:
   ...
```

Reply:
@SkalskiP I don't like the name triggers but that's what the corresponding variable is named on `develop`.
Lines: 
```
triggers = [
                self.vector.cross_product(point=anchor) < 0 for anchor in box_anchors
            ]

if len(set(triggers)) == 2:
    ...
```
Are replaced with: 
```
triggers = self._cross_product(all_anchors, self.vector) < 0
# Reduce to find out if all anchors for a
# detection lie to the left (or right) of the line
max_triggers = np.max(triggers, axis=0)
min_triggers = np.min(triggers, axis=0)
...

if min_triggers[i] != max_triggers[i]:
   ...
```

As you can see "new" `triggers` is just a vectorized version of the "old" `triggers`, i.e. matrix which holds cross product between an anchor and the vector for all `(anchor, detection)` pairs. Now, this cross product tells us whether a given anchor is "to the left"or "to the right" of the line. We update tracklet's state only if all its anchors are on the same side of the line. 
The following line from `develop` checks if all anchors of a detection are on the same side of the line:
```
if len(set(triggers)) == 2:
```
The lines you are referring to are a direct equivalent of this line for the vectorized version:
Triggers is a boolean array of shape `(number of anchors, detections)`.
If we want to know if all anchors for a detection with index `i` lie on the same side of the line we need to check that:
```
[triggers[anchor_1, i] == triggers[anchor_2, i] for (anchor_1, anchor_2) in possible_anchors]
```
Since `triggers` has at most two values (True / False) it's easier and more efficient to just calculate maximum and minimum value for each detection and then check if they are equal. 

What if I replaced: 
```
max_triggers = np.max(triggers, axis=0)
min_triggers = np.min(triggers, axis=0)

...

if min_triggers[i] != max_triggers[i]:
   ...
```

With something like: 
```
ANCHOR_AXIS = 0
max_triggers = np.max(triggers, axis=ANCHOR_AXIS)
min_triggers = np.min(triggers, axis=ANCHOR_AXIS)
all_anchors_on_the_same_side_of_line = max_triggers == min_triggers

...

if all_anchors_on_the_same_side_of_line[i]:
   ...
```
Would you find it more readable?

Reply:
@tc360950, thanks a lot for the explanation. I understand everything now.